src/PuliServiceProvider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the puli/silex-provider package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\SilexProvider;
+
+use Puli\GeneratedPuliFactory;
+use Puli\TwigExtension\PuliExtension;
+use Puli\TwigExtension\PuliTemplateLoader;
+use Silex\Application;
+use Silex\ServiceProviderInterface;
+
+class PuliServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * Registers services and parameters on the app.
+     *
+     * @param Application $app Silex application.
+     */
+    public function register(Application $app)
+    {
+        $app['puli.repository'] = $app->share(function (Application $app) {
+            return $app['puli.factory']->createRepository();
+        });
+
+        $app['puli.discovery'] = $app->share(function (Application $app) {
+            return $app['puli.factory']->createDiscovery($app['puli.repository']);
+        });
+
+        $app['puli.asset_url_generator'] = $app->share(function (Application $app) {
+            return $app['puli.factory']->createUrlGenerator($app['puli.discovery']);
+        });
+
+        // Templates loader
+        if (isset($app['twig'])) {
```

Comment:
The same was done for the Symfony bundle, since some people wanted to use it, but without changing Twig.